### PR TITLE
screen picker unit test

### PR DESCRIPTION
**Pull request by vrbsm**
## Description
- screen picker unit test
- screen sharing indicator unit test

## QA Checklist
- [x] Unit-Tests
- [ ] Automation-Tests

Attach unit & spectron tests results in PDF format against the above task lists for this branch
[Symphony Electron Test Result.pdf](https://github.com/symphonyoss/SymphonyElectron/files/2958205/Symphony.Electron.Test.Result.pdf)

### CODE REVIEW

File: spec/screenPicker.spec.ts [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+    it('should render correctly', () => {
+        const wrapper = shallow(React.createElement(ScreenPicker));
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it('should call `close` correctly', () => {
+        const { ipcRenderer } = require('./__mocks__/electron');
+        const wrapper = shallow(React.createElement(ScreenPicker));
+        const spy = jest.spyOn(ipcRenderer, sendEventLabel);
+        const customeSelector = 'button.ScreenPicker-cancel-button';
```
Comment (VishwasShashidhar) [inline spec/screenPicker.spec.ts]: typo. variable name should be `customSelector`

File: spec/screenPicker.spec.ts [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+        const expectedValue = { cmd: 'close-window', windowType: 'screen-picker' };
+        wrapper.find(customeSelector).simulate('click');
+        expect(spy).nthCalledWith(1, screenSourceSelectedLabel, null);
+        expect(spy).nthCalledWith(2, symphonyApiLabel, expectedValue);
+    });
+
+    it('should call `submit` correctly', () => {
+        const { ipcRenderer } = require('./__mocks__/electron');
```
Comment (VishwasShashidhar) [inline spec/screenPicker.spec.ts]: any reason why we are requiring this here?

File: spec/screenPicker.spec.ts [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+            ],
+            selectedSource: { display_id: '0', id: '0', name: 'Entire screen', thumbnail: undefined },
+        };
+        const applicationScreenStateMock = {
+            sources: [
+                { display_id: '0', id: '0', name: 'Application screen 0', thumbnail: undefined },
+            ],
+            selectedSource: { display_id: '0', id: '0', name: 'Application screen 0', thumbnail: undefined },
+        };
+        it('should call `onToggle` when screen tab is changed', () => {
```
Comment (VishwasShashidhar) [inline spec/screenPicker.spec.ts]: add a new line above this.

File: spec/screenPicker.spec.ts [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+            events.keyup(keyCode.endKey);
+            expect(spy).lastCalledWith(expectedValue);
+        });
+
+    });
+
+    describe('tab titles', () => {
+        it('should show `application-tab` when source name is not Entire screen', () => {
+            const wrapper = shallow(React.createElement(ScreenPicker));
+            const applicationScreenStateMock = { sources: [{ display_id: '1', id: '1', name: 'Application screen', thumbnail: undefined }]};
```
Comment (KiranNiranjan) [inline spec/screenPicker.spec.ts]: The use case where `Screen 2` is not at the beginning of the source name should not display the `screen tab`
```suggestion
            const applicationScreenStateMock = { sources: [
                   { display_id: '1', id: '1', name: 'Application screen', thumbnail: undefined }
                   { display_id: '2', id: '2', name: 'Application Screen 2', thumbnail: undefined }
            ]};
```

File: spec/screenPicker.spec.ts [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+            wrapper.setState(applicationScreenStateMock);
+            expect(wrapper.find(applicationTabCustomSelector)).toHaveLength(1);
+            expect(wrapper.find(screenTabCustomSelector)).toHaveLength(0);
+        });
+
+        it('should show `screen-tab` when source name is Entire screen', () => {
+            const wrapper = shallow(React.createElement(ScreenPicker));
+            const entireScreenStateMock = { sources: [{ display_id: '1', id: '1', name: 'Entire screen', thumbnail: undefined }]};
```
[the remark this answers is not in the document]
Comment (vrbsm) [reply spec/screenPicker.spec.ts]: :+1:

File: spec/screenPicker.spec.ts (hunk 4 above)
Comment (vrbsm) [reply spec/screenPicker.spec.ts]: :+1:

**Pull request merged**